fix: error message with stdout/stderr in the case of sharing command fails (#1984)
terramate-io/terramate: CHANGELOG.md

## Unreleased

### Fixed

> - Fix Outputs Sharing error handling by adding stdout and stderr context of the error.

Reviewer: I would make this less technical, for example just
```suggestion
- Improve the error reporting of the Outputs Sharing feature.
```
